External/FEXCore/Source/Interface/Core/Core.cpp

```diff
@@ -189,6 +192,48 @@ namespace FEXCore::Context {
     return NewThreadState;
   }
 
+  bool Context::HandleSegfault(FEXCore::Core::InternalThreadState *Thread, int Signal, void *info, void *ucontext) {
+    auto CTX = Thread->CTX;
+
+    auto FaultAddress = (uintptr_t)((siginfo_t*)info)->si_addr;
+
+    bool DoFlush = false;
+
+    {
+      FHU::ScopedSignalMaskWithSharedMutex lk(CTX->MemoryEntryMutex, true);
+
+      // Get the one after FaultAddress, or end
+      // FaultAddress is inclusive
+      // If the write spans two pages, they will be flushed one at a time (generating two faults)
+      auto Entry = CTX->MemoryMaps.upper_bound(FaultAddress);
+
+      if (Entry != CTX->MemoryMaps.begin()) {
+        --Entry;
+
+        if (Entry->first <= FaultAddress && (Entry->first + Entry->second.Length) > FaultAddress) {
+          if (Entry->second.Writable) {
+            DoFlush = true;
+          }
+        }
+      }
+    }
+
+    // Do FlushCodeRange outside MemoryEntryMutex lock to avoid CompileCode deadlock
```

Comment:
This sentence mixes logic summary with a side note. Better to separate the two:

```suggestion
    // Perform the actual FlushCodeRange if an overlapping mapping was found.
    // Note that this is done outside of the MemoryEntryMutex lock to avoid deadlocking in CompileCode
```